shotgun_api3/shotgun.py

```diff
@@ -3594,6 +3587,22 @@ def _make_call(self, verb, path, body, headers):
             attempt += 1
             try:
                 return self._http_request(verb, path, body, req_headers)
+            except ssl.SSLEOFError as e:
+                # SG-34910 - EOF occurred in violation of protocol (_ssl.c:2426)
+                # This issue seems to be related to proxy and keep alive.
+                # It looks like, sometimes, the proxy drops the connection on
+                # the TCP/TLS level despites the keep-alive. So we need to close
+                # the connection and make a new attempt.
+                LOG.debug("SSLEOFError: {}".format(e))
+                self._close_connection()
+                if attempt == max_rpc_attempts:
```

Comment:
`self.config.max_rpc_attempts` is hardcoding the value of `3`.  Maybe we can now get rid of this one and use `self.MAX_ATTEMPTS` instead. WDYT?

Reply:
Hardcoded yes and no. It's the default value of the config object but users can select a different value.

Also, I see that this variable is still used at other places. So I kind of want to keep it as is for the moment.